File: 0001-two-sum/0001-two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        nums = [(val, i)for i, val in enumerate(nums)]
        nums.sort(key=lambda x: x[0])
        
        left=0
        right=len(nums)-1

        while left<right:
            s= nums[right][0]+nums[left][0]
            if s==target:
                return nums[left][1], nums[right][1]
            if s<target:
                left+=1
            if s>target:
                right-=1
        return None

File: 0001-two-sum/0001-two-sum.cpp
#include <vector>
#include <algorithm>

class Solution {
public:
    std::vector<int> twoSum(std::vector<int>& nums, int target) {
        // 1. יצירת ווקטור של זוגות: {ערך, אינדקס מקורי}
        std::vector<std::pair<int, int>> indexedNums;
        for (int i = 0; i < nums.size(); ++i) {
            indexedNums.push_back({nums[i], i});
        }

        // 2. מיון לפי הערכים (האיבר הראשון בזוג)
        std::sort(indexedNums.begin(), indexedNums.end());

        // 3. שימוש בשני מצביעים על הווקטור הממוין
        int left = 0;
        int right = indexedNums.size() - 1;

        while (left < right) {
            int sum = indexedNums[left].first + indexedNums[right].first;

            if (sum == target) {
                // החזרת האינדקסים המקוריים שנשמרו בתוך הזוג
                return {indexedNums[left].second, indexedNums[right].second};
            }
            else if (sum < target) {
                left++;
            }
            else {
                right--;
            }
        }

        return {}; // במקרה שלא נמצא פתרון
    }
};